tjpcov_new/covariance_clusters.py

```diff
@@ -0,0 +1,632 @@
+                            * self.double_bessel_integral(z1,vec_final[m])\
+                            * self.integral_mass(vec_final[m],bin_lbd_j)
+                factor_approx = 1 
+
+        return (romb(kernel, dx=romb_range)) * factor_approx
+
+
+    def double_bessel_integral(self, z1, z2):
+        """Calculates the double bessel integral from I-ell algorithm, as function of z1 and z2
+        """
+
+        # definition of t, forcing it to be <= 1
+        r1 = ccl.comoving_radial_distance(self.cosmo, 1/(1+z1))
+        r2 = r1
+        R = 1
+        if z1 != z2:
+            r2 = ccl.comoving_radial_distance(self.cosmo, 1/(1+z2))
+            R = min(r1, r2)/max(r1, r2)
+
+        I_ell_vec = [self.I_ell(m, R ) for m in range(self.N//2+1)]
+
+        back_FFT_vec = np.fft.irfft(self.Phi_vec*I_ell_vec)*self.N  # FFT back
+        two_fast_vec = (1/np.pi)*(self.ko**3)*((self.r_vec/self.ro) ** (-self.bias_fft))*back_FFT_vec/self.G
+
+        imin = self.imin
+        imax = self.imax
+       
+        # we will use this to interpolate the exact r(z1)
+        f = interp1d(self.r_vec[imin:imax], two_fast_vec[imin:imax], kind='cubic') 
+        try:
+            return f(max(r1, r2))
+        except Exception as err:
+            print(err,f"""\nValue you tried to interpolate: {max(r1,r2)} Mpc, 
+                Input r {r1}, {r2}
+            Valid range range: [{self.r_vec[self.imin]}, {self.r_vec[self.imax]}] Mpc""")
+
+        #CHECK THE INDEX NUMBERS
+        # TODO test interpolkind
+   
+    def eval_sigma_vec(self):
+        sigma_vec = np.zeros((self.n_z, self.n_z))
+
+        for i in range (self.n_z):
+            for j in range (i, self.n_z):
+
+                sigma_vec[i,j] = self.double_bessel_integral(self.z_true_vec[i],self.z_true_vec[j])
+                sigma_vec[j,i] = sigma_vec[i,j]
+
+        return sigma_vec
+    
+    def get_min_radial_idx(self):
+        return np.argwhere(self.r_vec < 0.95*self.radial_lower_limit)[-1][0]
+    
+    def get_max_radial_idx(self):
+        return np.argwhere(self.r_vec  > 1.05*self.radial_upper_limit)[0][0]
+
+
+
+class CovarianceClusterCounts(CovarianceClusters):
+    # Figure out
+    # cov_type = 'gauss'
+    # _reshape_order = 'F'
+
+    def __init__(self, config):
+        super().__init__(config)
+
+        # Find out correct way to pull from sacc
+        self.ssc_conf = self.config.get('SSC', {})
+
+        romberg_num = 2**6+1
+        self.Z1_true_vec = np.zeros((self.num_z_bins, romberg_num))
+        self.G1_true_vec = np.zeros((self.num_z_bins, romberg_num))
+        self.dV_true_vec = np.zeros((self.num_z_bins, romberg_num))
+        self.M1_true_vec = np.zeros((self.num_richness_bins, self.num_z_bins, romberg_num))
+    
+    def get_covariance_block(self, tracer_comb1=None, tracer_comb2=None,
+                             integration_method=None,
+                             include_b_modes=True):
+        
```

Comment:
I'm a little confused here about how to incorporate the tracers.  Our cluster count tracers are currently formatted as

clusters_0_1
clusters_0_2
...
clusters_3_1
clusters_3_2
...
etc

I think I can just ignore these options, since theyre optional anyway, right?